snap7/util.py

```diff
@@ -642,25 +644,35 @@ def set_time(bytearray_: bytearray, byte_index: int, time_string: str) -> bytear
 
         Examples:
             >>> data = bytearray(4)
-            >>> snap7.util.set_dint(data, 0, '-22:3:57:28.192')
+
+            >>> snap7.util.set_time(data, 0, '-22:3:57:28.192')
+
             >>> data
                 bytearray(b'\x8d\xda\xaf\x00')
         """
-    import re
     sign = 1
     bits = 32
     data_list = re.split('[: .]', time_string)
-    days, hours, minutes, seconds, milli_seconds = [int(x) for x in data_list]
-    if days < 0:
+    days: str = data_list[0]
+    hours: int = int(data_list[1])
+    minutes: int = int(data_list[2])
+    seconds: int = int(data_list[3])
+    milli_seconds: int = int(data_list[4])
+    if re.split(r'(\d+)$', days)[0:2][0] == '-':
         sign = -1
-    time_int = ((days * sign * 3600 * 24 + (hours % 24) * 3600 + (minutes % 60) * 60 + seconds % 60) * 1000 + milli_seconds) * sign
-    if sign < 0:
-        time_int = (1 << bits) + time_int
-    formatstring = '{:0%ib}' % bits
-    byte_hex = hex(int(formatstring.format(time_int), 2)).split('x')[1]
-    bytes_array = bytes.fromhex(byte_hex)
-    bytearray_[byte_index:byte_index + 4] = bytes_array
-    return bytearray_
+    if abs(int(days)) <= 24 and hours <= 23 and minutes <= 59 and seconds <= 59 and milli_seconds <= 999:
+        if int(days) * sign == 24 and hours > 20 and minutes > 31 and seconds > 23 and milli_seconds > 647 or \
+                int(days) * sign == -24 and hours > 20 and minutes > 31 and seconds > 23 and milli_seconds > 648:
+            raise ValueError('time value out of range, please check the value interval')
+
+        time_int = ((int(days) * sign * 3600 * 24 + (hours % 24) * 3600 + (minutes % 60) * 60 + seconds % 60) * 1000 + milli_seconds) * sign
+        if sign < 0:
+            time_int = (1 << bits) + time_int
+        bytes_array = time_int.to_bytes(4, byteorder='big')
```

Comment:
The sign check can be replaced by passing a parameter `signed=True` to the method `.to_bytes()`.
```suggestion
        bytes_array = time_int.to_bytes(4, byteorder='big', signed=True)
```